code/SingleObjectAdmin.php

```diff
@@ -174,13 +174,13 @@ public function doSave($data, $form)
             }
         }
 
-        $link = '"' . $object->singular_name() . '"';
+        $link = '"' . $object->i18n_singular_name() . '"';
         $message = _t(
             'GridFieldDetailForm.Saved',
             'Saved {name} {link}',
             array(
-                'name' => $object->singular_name(),
-                'link' => $link
+                'name' => $object->i18n_singular_name(),
+                'link' => ''
```

Comment:
This _t() function refers to `GridFieldDetailForm.Saved` in the framework lang file *framework/lang/nl.yml* which has this value:

    Saved: '{name} {link} opgeslagen'

Removing *'link' => ''* from the array would leave a meaningless **DataObject saved {link}** in the successmessage, whereas an empty value for 'link' doesn't leave any marks. If we would remove *{link}* from the array we would need to remove it from the framework lang file which isn't desired obviously.